android/src/main/java/com/reactnativecommunity/webview/RNCWebViewManager.java

```diff
@@ -1096,16 +1102,38 @@ public void run() {
             }
             WritableMap data = mRNCWebViewClient.createWebViewEvent(webView, webView.getUrl());
             data.putString("data", message);
-            dispatchEvent(webView, new TopMessageEvent(webView.getId(), data));
+
+            if (reactContext != null) {
+              mContext.sendDirectMessage(reactContext, data);
+            } else {
+              dispatchEvent(webView, new TopMessageEvent(webView.getId(), data));
+            }
           }
         });
       } else {
         WritableMap eventData = Arguments.createMap();
         eventData.putString("data", message);
-        dispatchEvent(this, new TopMessageEvent(this.getId(), eventData));
+
+        if (reactContext != null) {
+          this.sendDirectMessage(reactContext, eventData);
+        } else {
+          dispatchEvent(this, new TopMessageEvent(this.getId(), eventData));
+        }
       }
     }
 
+    protected void sendDirectMessage(ReactContext reactContext, WritableMap data) {
+      CatalystInstance catalystInstance = reactContext.getCatalystInstance();
```

Comment:
Can the instance be a class variable? To avoid initializing this instance ever time onMessage is called.

Reply:
Changed accordingly.